polkadot/xcm/xcm-builder/src/location_conversion.rs

```diff
@@ -317,6 +317,22 @@ impl<Network: Get<Option<NetworkId>>, AccountId: From<[u8; 32]> + Into<[u8; 32]>
 	}
 }
 
+/// Extracts the `AccountId32` from the passed Treasury plurality if the network matches.
+pub struct TreasuryVoiceConvertsVia<Network, AccountId>(PhantomData<(Network, AccountId)>);
+impl<Network: Get<Option<NetworkId>>, AccountId: From<[u8; 32]> + Into<[u8; 32]> + Clone>
+	ConvertLocation<AccountId> for TreasuryVoiceConvertsVia<Network, AccountId>
+{
+	fn convert_location(location: &MultiLocation) -> Option<AccountId> {
+		let id: [u8; 32] = match *location {
+			MultiLocation { parents: 0, interior: X1(Plurality { id: BodyId::Treasury, .. }) } |
+			MultiLocation { parents: 1, interior: X1(Plurality { id: BodyId::Treasury, .. }) } =>
+				PalletId(*b"py/trsry").into_account_truncating(),
```

Comment:
IMO this should be generic. `TreasuryAccount: Get<AccountId>`

Reply:
Sure I can do that, shall I keep the function in this file or move it somewhere else?

Reply:
Which function?

Reply:
disregard, i used TreasuryAccount and passing it in from the runtime/xcm_config